### DISCUSSION TITLE
How can I pass some data to my custom tool JS function without including it in the 'question'?

### DISCUSSION THREAD

**Discussion by bhowerton** [FlowiseAI/Flowise#990] [Q&A]
When POSTing to the flowise completion endpoint, I'd like to include some user-specific information that is then accessible within the Javascript function of a custom tool. I'd rather not include it in the question since I don't want the model to consider it in its response.  

Ex: Include a guid from the calling application that is then passed on to an API call made within a custom tool JS function.

**Comment by HenryHengZJ** [comment]
interesting idea, can look at how to do that. was thinking of allowing user to pass in global env var, which then can be used in custom func

**Reply by vertgo** [reply]
how about a general json node?

**Reply by HenryHengZJ** [reply]
can now this via global env vars - https://docs.flowiseai.com/using-flowise/variables
